src/redux/reducer.ts

```diff
@@ -1737,24 +1733,44 @@ export const reducer = createReducer(initialState, (builder) => {
         state.tableSort[SPREADSHEET_SORT_STORE][newTabName] = value;
     });
 
-    builder.addCase(UPDATE_CUSTOM_COLUMNS_DEFINITION, (state, action: UpdateCustomColumnsDefinitionsAction) => {
-        state.tables.allCustomColumnsDefinitions[action.colWithFormula.index] =
-            state.tables.allCustomColumnsDefinitions[action.colWithFormula.index].some(
-                (column) => column.uuid === action.colWithFormula.value.uuid
-            )
-                ? state.tables.allCustomColumnsDefinitions[action.colWithFormula.index].map((column) =>
-                      column.uuid === action.colWithFormula.value.uuid ? action.colWithFormula.value : column
-                  )
-                : [
-                      ...state.tables.allCustomColumnsDefinitions[action.colWithFormula.index],
-                      action.colWithFormula.value,
-                  ];
-    });
-
-    builder.addCase(REMOVE_CUSTOM_COLUMNS_DEFINITION, (state, action: RemoveCustomColumnsDefinitionsAction) => {
-        state.tables.allCustomColumnsDefinitions[action.definition.index] = state.tables.allCustomColumnsDefinitions[
-            action.definition.index
-        ].filter((column) => column.id !== action.definition.value);
+    builder.addCase(UPDATE_COLUMNS_DEFINITION, (state, action: UpdateColumnsDefinitionsAction) => {
+        const { colData } = action;
+
+        // Retrieve the table definition by index
+        const tableDefinitionIndex = state.tables.definitions.findIndex((def) => def.index === colData.index);
+
+        if (tableDefinitionIndex !== -1) {
+            const existingColumnIndex = state.tables.definitions[tableDefinitionIndex].columns.findIndex(
+                (col) => col.id === colData.value.id
+            );
```

Comment:
```suggestion
        const tableDefinition = state.tables.definitions[colData.index];
        if (tableDefinition) {
            const existingColumnIndex = tableDefinition.columns.findIndex((col) => col.id === colData.value.id);
```
Why not this directly ?